### PR TITLE
Messaging and concurrency fixes

### PR DESCRIPTION
**Pull request by bethrichardson**
Includes:
- Messages are sent in JSON format and with an enumerated MessageType.
- Added send message threads for nodes
- Squashed some messaging mishandling
- Algorithms correctly wait for values before proceeding through phases.
- Divided algorithms into phases to better show how phases are operating.
- nodes are now initiated separately from coordinator
- Coordinator now uses a better format for determining consensus based, not yet on weights, but on a 3/4 group wins.
- Began work on determine faulty algorithm to better match WBA paper.
- All tests pass except for faulty node determination, which still needs some debugging to get right.

### CODE REVIEW

File: src/ConsensusAlgorithm.java [hunk 1]
```diff
@@ -1,6 +1,13 @@
+import com.sun.org.apache.xpath.internal.operations.Bool;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
```
Comment (recreationalcode) [inline src/ConsensusAlgorithm.java]: Are these imports being used? You can clean up unused imports in IntelliJ, that way they don't get pulled in unnecessarily

File: src/Utils.java [hunk 2]
```diff
@@ -1,8 +1,12 @@
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
```
Comment (recreationalcode) [inline src/Utils.java]: Unused imports?

**Comment by recreationalcode** [comment]
Can we add maven support to this project? It's becoming very difficult to manage dependencies, especially with poor internet connection

File: src/Utils.java (hunk 2 above)
Comment (bethrichardson) [inline src/Utils.java]: Sorry yea I usually optimize imports. Will fix next time I'm on.

**Comment by bethrichardson** [comment]
Feel free to submit a pull request with the maven support. I can try to do it next time I'm working on the project too. Let me know. @fobofindia09

**Pull request merged**